src/System.Private.ServiceModel/src/System/IdentityModel/Tokens/SamlSerializer.cs

```diff
@@ -2,15 +2,250 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
 using System.IdentityModel;
+using System.IdentityModel.Selectors;
+using System.Runtime.Serialization;
+using System.Security;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.ServiceModel;
+using System.Text;
+using System.Xml;
+using DiagnosticUtility = System.ServiceModel.DiagnosticUtility;
 
 namespace System.IdentityModel.Tokens
 {
     public class SamlSerializer
     {
+        private DictionaryManager _dictionaryManager;
+
         public SamlSerializer()
         {
         }
+
+        // Interface to plug in external Dictionaries. The external
+        // dictionary should already be populated with all strings 
+        // required by this assembly.
+        public void PopulateDictionary(IXmlDictionary dictionary)
+        {
+            if (dictionary == null)
+                throw ServiceModel.DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("dictionary");
+
+            _dictionaryManager = new DictionaryManager(dictionary);
+        }
+
+        internal DictionaryManager DictionaryManager
+        {
+            get
+            {
+                if (_dictionaryManager == null)
+                    _dictionaryManager = new DictionaryManager();
+
+                return _dictionaryManager;
+            }
+        }
+
+        // Issue #31 in progress
```

Comment:
I'm not expecting you to go back and change everything, this is just an idea, but #if'ing the blocks out might make things easier as you can temporarily define the item to see how much left there is to resolve.